src/main/java/org/folio/circulation/domain/validation/ChangeDueDateValidator.java

```diff
@@ -17,33 +20,32 @@ public ChangeDueDateValidator(LoanRepository loanRepository) {
     this.loanRepository = loanRepository;
   }
 
-  public CompletableFuture<Result<LoanAndRelatedRecords>> refuseWhenClaimedReturned(
-      Result<LoanAndRelatedRecords> changedLoanResult) {
+  public CompletableFuture<Result<LoanAndRelatedRecords>> refuseChangeDueDateForItemInStatus(
+    Result<LoanAndRelatedRecords> loanAndRelatedRecordsResult) {
 
-    return changedLoanResult
-      .map(LoanAndRelatedRecords::getLoan)
-      .afterWhen(this::itemIsClaimedReturned,
-        this::refuseWhenDueDateChanged, this::carryOn)
-      .thenApply(r -> r.next(existingLoan -> changedLoanResult));
+    return loanAndRelatedRecordsResult.after(context -> refuseWhenDueDateChanged(context,
+      Loan::isDeclaredLost,
+      Loan::isClaimedReturned,
+      Loan::isAgedToLost));
   }
 
-  private CompletableFuture<Result<Boolean>> itemIsClaimedReturned(
-      Loan changedLoan) {
+  @SafeVarargs
+  private final CompletableFuture<Result<LoanAndRelatedRecords>> refuseWhenDueDateChanged(
+    LoanAndRelatedRecords relatedRecords, Predicate<Loan>... refusePredicates) {
```

Comment:
Why does this method need to check the item status when `ItemStatusValidator. refuseWhenItemIsNotAllowedForDueDateChange` checks those statuses as well?

Is it because the item status validator is not used when the due date is changed during a PUT to a loan resource?

If so, could the change due date validator use the item status validator method?

Reply:
Yes, this validator is only used in loan PUT.

Reply:
The status validator method is now being reused.